### PR TITLE
Selenium tests

### LINKED ISSUE #49: Setup general testing infrastructure for frontend
Old: Don't call 'cl' to create temporary worksheet.

Split this up into multiple issues.

### PR DESCRIPTION
**Pull request by yashsavani**
Fixes #49 Working Selenium testing infrastructure with 9 tests

### CODE REVIEW

**Comment by kashizui** [comment]
Commandeering this PR.
Some notes so far:

- Requires installing geckodriver
- Relies on specific order of the test methods, which is not default in unittest.

**Comment by kashizui** [comment]
@percyliang Finished fixing the tests, here's a video of a test run:

http://web.stanford.edu/~sckoo/push/YZtcFBr7udflR72Y.mov

Please review

File: tests/README.md [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
+
+        cd path/to/codalab-worksheets
+
+1. Ensure that you have selenium installed.
+
+        ./setup.sh
+
+2. Download and install [Mozilla geckodriver](https://github.com/mozilla/geckodriver). On macOS:
+
+        brew install geckodriver
```
Comment (percyliang) [inline tests/README.md]: For Linux, add instructions: wget https://github.com/mozilla/geckodriver/releases/download/v0.13.0/geckodriver-v0.13.0-linux64.tar.gz, unzip and put `geckodriver` into the path.

File: tests/test.py [hunk 2]
```diff
@@ -1,146 +1,214 @@
...
+        self.worksheet = self.username + '-testing'
+
+    def step1_title(self):
+        """Tests that the title of the page is as expected"""
+        self.driver.get("http://localhost:8000")
+        self.assertEqual(
+            'CodaLab - Home',
+            self.driver.title)
+
+    def step2_sign_up(self):
```
Comment (percyliang) [inline tests/test.py]: Can we skip this step?  (Assume that the user is just already signed in).  We can have sign up as another test that's not the default...the problem is that when I signed up on my account, the email sending was on, so it sent the confirmation email to some random @abc.com account.

File: tests/test.py [hunk 3]
```diff
@@ -1,146 +1,214 @@
...
+    def get_testing_worksheet(self):
+        """Internal function that navigates to the new testing worksheet"""
+        self.driver.get("http://localhost:8000/worksheets/%s/" % self.worksheet_uuid)
+
+    def step5_upload_file(self):
+        """Tests that we can upload a.txt and b.txt"""
+        self.get_testing_worksheet()
+        self.driver.find_element_by_partial_link_text("Upload").click()
+        raw_input("Go to the browser window and select file "
+                  "codalab-worksheets/tests/a.txt to upload, then press enter")
```
Comment (percyliang) [inline tests/test.py]: This was a bit cumbersome, but I don't see a way around this.

File: tests/test.py [hunk 4]
```diff
@@ -1,146 +1,214 @@
...
+    def step1_title(self):
+        """Tests that the title of the page is as expected"""
+        self.driver.get("http://localhost:8000")
+        self.assertEqual(
+            'CodaLab - Home',
+            self.driver.title)
+
+    def step2_sign_up(self):
+        """Tests that users can sign up as expected"""
+        self.driver.get("http://localhost:8000")
```
Comment (percyliang) [inline tests/test.py]: We want to be able to point this to worksheets.codalab.org or worksheets-test.codalab.org, so make this a command-line argument.

File: tests/test.py (hunk 2 above)
Comment (percyliang) [inline tests/test.py]: We can also prompt the user for username/password when the test.py script runs and use that.

**Comment by kashizui** [comment]
TODO

- [x] Document geckodriver installation for linux
- [ ] Skip signup test
- [ ] Make host a command-line arg
- [ ] Assume user is already signed in, or prompt for username/password

**Comment by kashizui** [comment]
@percyliang Made the requested changes, please see demo below and review!

- made upload process smoother (no need to press enter on prompt)
- disabled sign up test
- prompt for host, username, password (decided not to do command-line arguments because they would conflict with the command line arguments that `unittest` takes.)
- clean up the created bundles and worksheet at the end

http://web.stanford.edu/~sckoo/push/965orxy4Hw9ovh9B.mov

**Comment by percyliang** [comment]
LGTM

**Pull request merged**